### PR TITLE
Deploy resources in kind order

### PR DESCRIPTION
**Pull request by adammw**
Deploy resources in kind order specified by https://github.com/helm/helm/blob/release-2.13/pkg/tiller/kind_sorter.go#L29

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: https://zendesk.atlassian.net/browse/PAAS-3365

### Risks
- Level: Low/Med/High

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/release_doc.rb [hunk 1]
```diff
@@ -43,12 +43,15 @@ def resource_template
...
-        )
+      @resources ||= begin
+        resources = resource_template.map do |t|
+          Kubernetes::Resource.build(
+            t, deploy_group,
+            autoscaled: kubernetes_role.autoscaled,
+            delete_resource: delete_resource
+          )
+        end
+        resources.sort_by { |r| Kubernetes::RoleConfigFile::DEPLOY_SORT_ORDER.index(r.kind) }
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/release_doc.rb]:
```suggestion
        resources.sort_by { |r| Kubernetes::RoleConfigFile::DEPLOY_SORT_ORDER.index(r.kind) || 999 }
```

File: plugins/kubernetes/app/models/kubernetes/release_doc.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class ReleaseDoc < ActiveRecord::Base
     attr_reader :previous_resources
 
     def deploy
-      @previous_resources = resources.map(&:resource)
+      @previous_resources = resources.reverse.map(&:resource)
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/release_doc.rb]: We'd still want to deploy the configmaps before we deploy anything else etc ... so keep this ?

**Comment by grosser** [comment]
the deploy_executor still deploys them in parallel, need to remove that too deploy_executor.rb:370 :(

File: plugins/kubernetes/app/models/kubernetes/release_doc.rb (hunk 1 above)
Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/release_doc.rb]: used the size instead of a hardcoded number

**Comment by adammw** [comment]
Is it worth partitioning by kind and doing parallel per kind or is that needless premature optimisation?

**Review by uthark — APPROVED** [review]

**Comment by grosser** [comment]
premature ... we don't deploy that many things of the same kind in parallel

**Review by grosser — APPROVED** [review]
kinda sad that we mess with the oder the users gave us, but if that's what kubectl does then 🤷‍♂

**Comment by adammw** [comment]
i mean, if it was doing it in parallel then the order would be messed up anyway..

**Comment by grosser** [comment]
True 😄

**Pull request merged**